controllers/metallb_controller_test.go

```diff
@@ -0,0 +1,56 @@
+package controllers
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/metallb/metallb-operator/api/v1alpha1"
+	"github.com/metallb/metallb-operator/test/consts"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("MetalLB Controller", func() {
+	Context("syncMetalLB", func() {
+		speakerImage := "test-speaker:latest"
+		controllerImage := "test-controller:latest"
+		BeforeEach(func() {
+			By("Setting the environment variables")
+			Expect(os.Setenv("SPEAKER_IMAGE", speakerImage)).To(Succeed())
+			Expect(os.Setenv("CONTROLLER_IMAGE", controllerImage)).To(Succeed())
+		})
+		It("Should create manifests with the correct values", func() {
+			By("Creating a Metallb resource")
+			metallb := &v1alpha1.Metallb{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "metallb",
+					Namespace: consts.MetallbNameSpace,
+				},
+			}
+			err := k8sClient.Create(context.Background(), metallb)
```

Comment:
Should we clean the metallb after the test?
This would be handy if more tests are added

Reply:
done